src/lib/util.ts

```diff
@@ -65,16 +65,46 @@ function shouldOverrideEvent(overrideDict: { [index: string]: boolean }, eventTy
   return false
 }
 
+function hashMapToString (obj: { [key: string] : string }): string {
+  let result = ''
+  Object.entries(obj).forEach(kv => {
+    const [k, v] = kv
+    result = result + `${k}:${v};`
+  })
+  return result.slice(0, -1)
+} 
+
+const getOwnTracestateMap = (vendor: string, tracestate: string): { [key: string]: string, _rest: string } => {
+  if (!tracestate.includes(vendor)) {
+    return { _rest: tracestate }
+  }
+  const stateHashMap: { [key: string]: string, _rest: string } = { _rest: '' }
+  const t = tracestate
+    .split(',') // get each vendor
+    .filter(ts => ts.split('=')[0] === vendor)[0] //get own vendor
```

Comment:
Rather check the key of the key-value to make sure its the "spanId".

Reply:
this is getting our tags as hash map and the rest as the string it is, yet.
so what the check would give us exactly? just to make sure we have it?